Add function to manually set or get auth route token

colin-oos opened the pull request:
This PR is in reference to issue #58 and allows users of authenticated routes to manually add or retrieve their access token through the javascript SDK. This is particularly useful for OAuth clients that authenticate with an existing access token rather than a `client_id:client_secret` pair. 

Adds the following methods:

```
api.auth.setToken(ACCESS_TOKEN);
console.log(ACCESS_TOKEN === api.auth.token);
// output: true
```

colin-oos committed:
Add function to manually set or get auth route token

hutch78 commented:
Hi @colin-oos 

Thank you for your patience and willingness to help make the `boxcast-sdk-js` better! Just out of curiosity, what steps are you taking to get the token prior to calling `setToken()`?

I like your changes. Would you mind adding a test in `api-authenticated.spec.ts` to cover the new methods? Something like:
```
  it('should allow the token to be manually set', async () => {
    api.auth.setToken(FAKE_TOKEN);
    expect(api.auth.token).toEqual(FAKE_TOKEN);
  });
```

I will keep a close eye on this to get it packaged up and published as soon as it's merged. Thanks again Colin!

colin-oos committed:
Add automated test for setting the auth token and update distribution

colin-oos commented:
@hutch78 I've added the test and also updated the distribution folder. I found I had to do that in order for the test to pick up the new `setToken` method, though I can revert the changes in the `dist` directory if you'd like.
